explore/Classifier_results-different_genes.py

```diff
@@ -0,0 +1,291 @@
+
+# coding: utf-8
+
+# # Create a logistic regression model to predict gene mutation from gene expression data in TCGA
+
+# In[1]:
+
+import os
+import urllib
+import random
+import warnings
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing, grid_search
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+
+
+# In[2]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[3]:
+
+# Parameter Sweep for Hyperparameters
+n_feature_kept = 500
+param_fixed = {
+    'loss': 'log',
+    'penalty': 'elasticnet',
+}
+param_grid = {
+    'alpha': [10 ** x for x in range(-6, 1)],
+    'l1_ratio': [0, 0.05, 0.1, 0.2, 0.5, 0.8, 0.9, 0.95, 1],
+}
+
+
+# *Here is some [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) regarding the classifier and hyperparameters*
+# 
+# *Here is some [information](https://ghr.nlm.nih.gov/gene/TP53) about TP53*
+
+# ## Load Data
+
+# In[4]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('download', 'expression-matrix.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[5]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('download', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[6]:
+
+# GENES
+
+genes_LungCancer = {
+    '207': 'AKT1', 
+    '238': 'ALK',  
+    '673': 'BRAF', 
+    '4921':'DDR2',
+    '1956':'EGFR',
+    '2064':'ERBB2',
+    '3845':'KRAS',
+    '5604':'MAP2K1',
+    '4893':'NRAS',
+    '5290':'PIK3CA',
+    '5728':'PTEN',
+    '5979':'RET',
+    '6016':'RIT1',
+    '6098':'ROS1',
+}
+
+genes_TumorSuppressors = {
+    '324': 'APC',  
+    '672': 'BRCA1',  
+    '675': 'BRCA2',
+    '1029':'CDKN2A',
+    '1630':'DCC',
+    '4089':'SMAD4',
+    '4087':'SMAD2',
+    '4221':'MEN1',
+    '4763':'NF1',
+    '4771':'NF2',
+    '7157':'TP53', 
+    '5728':'PTEN', 
+    '5925':'Rb',
+    '7428':'VHL',
+    '7486':'WRN',
+    '7490':'WT1',
+}
+
+genes_Oncogenes = {
+    '5155':'PDGFB', #growth factor
+    '5159':'PDGFRB', #growth factor 
+    '3791':'KDR', #receptor tyrosine kinases
+    '25':'ABL1', #Cytoplasmic tyrosine kinases
+    '6714':'SRC', #Cytoplasmic tyrosine kinases
+    '5894':'Raf-1',#cytoplasmic serine kinases

```

Comment:
The [primary symbol](https://www.ncbi.nlm.nih.gov/gene/5894) for this gene is RAF1. In general human gene names are all upercase.